### PR TITLE
[12/N] Fix clang-tidy warnings in jit

### PR DESCRIPTION
**Pull request by cyyever**
Follows #132131

cc @EikanWang @jgong5

### CODE REVIEW

**Review by Skylion007 — COMMENTED** [review]
Some nite and concerns. Double check that some of the const refs shouldn't be moved by value. I suppose updating mutate to const ref should be fine since it's all pass by value initially.

File: torch/csrc/jit/tensorexpr/ir_simplifier.h [hunk 1]
```diff
@@ -114,28 +112,25 @@ inline ExprPtr evaluateOp(ExprPtr v) {
...
   Term(HashProvider& hasher, ExprPtr s, Args... ts)
       : ExprNodeBase(promoteTypesVar(s, ts...)), scalar_(s), hasher_(hasher) {
     CHECK(s->isConstant());
     addComponent(ts...);
     sort();
   }
 
-  // NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init)
-  Term(HashProvider& hasher, ExprPtr s, std::vector<ExprPtr> v)
+  Term(HashProvider& hasher, const ExprPtr& s, const std::vector<ExprPtr>& v)
```
[the remark this answers is not in the document]
Comment (cyyever) [reply torch/csrc/jit/tensorexpr/ir_simplifier.h:122]: The vector is iterated but not assigned.

File: torch/csrc/jit/tensorexpr/ir_simplifier.h [hunk 2]
```diff
@@ -381,12 +376,12 @@ class TORCH_API SimplifierUnderContext : public IRMutator {
...
   ~SimplifierUnderContext() override = default;
   // Add boundary info for index variables in for-loops
-  StmtPtr mutate(ForPtr v) override;
+  StmtPtr mutate(const ForPtr& v) override;
 
-  ExprPtr mutate(DivPtr v) override;
-  ExprPtr mutate(ModPtr v) override;
-  ExprPtr mutate(CompareSelectPtr v) override;
-  ExprPtr mutate(IfThenElsePtr v) override;
+  ExprPtr mutate(const DivPtr& v) override;
```
[the remark this answers is not in the document]
Comment (cyyever) [reply torch/csrc/jit/tensorexpr/ir_simplifier.h:381]: v->lhs() and v->rhs() are called, both are const methods.

File: torch/csrc/jit/tensorexpr/ir_simplifier.h [hunk 3]
```diff
@@ -114,28 +112,25 @@ inline ExprPtr evaluateOp(ExprPtr v) {
...
         hasher_(hasher) {
     sort();
   }
 
   // Convenience constructor from a map of hash -> var, used when merging Terms.
-  // NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init)
   Term(
       HashProvider& hasher,
-      ExprPtr s,
+      const ExprPtr& s,
```
Comment (Skylion007) [inline torch/csrc/jit/tensorexpr/ir_simplifier.h:133]: shouldn't we just use `scalar_` for anything we want to reference in the ctor (and use std::move to move it into scalar_)?

File: torch/csrc/jit/tensorexpr/ir_simplifier.h [hunk 4]
```diff
@@ -119,10 +119,10 @@ class Term : public ExprNode<Term> {
     sort();
   }
 
-  Term(HashProvider& hasher, const ExprPtr& s, const std::vector<ExprPtr>& v)
+  Term(HashProvider& hasher, ExprPtr s, std::vector<ExprPtr> v)
       : ExprNodeBase(promoteTypesVec(s, v)),
```
Comment (Skylion007) [inline torch/csrc/jit/tensorexpr/ir_simplifier.h:123]: Eh, are these safe with promotTypesVec, I don't think it's actually defined that the initializers run in order, what you had before was right.

Comment (Skylion007) [reply torch/csrc/jit/tensorexpr/ir_simplifier.h:123]: Ah sorry, I missed this line when reviewing the PR on my phone, what you had before was right.

**Comment by Skylion007** [comment]
Sorry, the ctor for Term was actually correct before my comments. The updated `For` ctor looks good though.

**Comment by cyyever** [comment]
> Sorry, the ctor for Term was actually correct before my comments. The updated `For` ctor looks good though.

The current change seems right.

**Review by Skylion007 — APPROVED** [review]

**Pull request closed without merging**